transcode/transcode.go

```diff
@@ -120,20 +129,39 @@ func RunTranscodeProcess(transcodeRequest TranscodeSegmentRequest, streamName st
 	}
 
 	// Build the manifests and push them to storage
-	manifestManifestURL, err := GenerateAndUploadManifests(sourceManifest, targetTranscodedRenditionOutputURL.String(), transcodedStats)
+	manifestURL, err := GenerateAndUploadManifests(sourceManifest, targetTranscodedRenditionOutputURL.String(), transcodedStats)
 	if err != nil {
 		return outputs, segmentsCount, err
 	}
 
-	output := clients.OutputVideo{Type: "object_store", Manifest: manifestManifestURL}
+	playbackBaseURL := PublishDriverSession(targetTranscodedRenditionOutputURL.String(), tout.Path)
+
+	manifestURL = strings.ReplaceAll(manifestURL, targetTranscodedRenditionOutputURL.String(), playbackBaseURL)
+	output := clients.OutputVideo{Type: "object_store", Manifest: manifestURL}
 	for _, rendition := range transcodedStats {
-		output.Videos = append(output.Videos, clients.OutputVideoFile{Location: rendition.ManifestLocation, SizeBytes: int(rendition.Bytes)})
+		videoManifestURL := strings.ReplaceAll(rendition.ManifestLocation, targetTranscodedRenditionOutputURL.String(), playbackBaseURL)
+		output.Videos = append(output.Videos, clients.OutputVideoFile{Location: videoManifestURL, SizeBytes: int(rendition.Bytes)})
 	}
 	outputs = []clients.OutputVideo{output}
 	// Return outputs for .dtsh file creation
 	return outputs, segmentsCount, nil
 }
 
+func PublishDriverSession(osUrl string, relPath string) string {
+	osDriver, err := drivers.ParseOSURL(osUrl, true)
+	if err != nil {
+		return ""
+	}
+	videoUrl, err := osDriver.Publish(context.Background())
+	if err == drivers.ErrNotSupported {
+		return osUrl
+	} else if err == nil {
+		res, _ := url.JoinPath(videoUrl, relPath)
+		return res
+	}
+	return ""
```

Comment:
What does this default case (i.e we get some error but not the error about Publish being unsupported) mean here? Could you add an explanatory comment to the method please (also to explain about when Publish is / isn't supported for people unfamiliar for the w3s work)?

Reply:
Added a comment.